api/proxy/store/eigenda_manager.go

```diff
@@ -236,47 +294,19 @@ func (m *EigenDAManager) putToCorrectEigenDABackend(ctx context.Context, value [
 	return nil, fmt.Errorf("unsupported dispersal backend: %v", backend)
 }
 
-func (m *EigenDAManager) getFromCorrectEigenDABackend(
-	ctx context.Context,
-	versionedCert certs.VersionedCert,
-	opts common.GETOpts,
-) ([]byte, error) {
-	switch versionedCert.Version {
-	case certs.V0VersionByte:
-		m.log.Debug("Reading blob from EigenDAV1 backend")
-
-		if opts.ReturnEncodedPayload {
-			return nil, fmt.Errorf("returning encoded payload is not supported for V0 certificates (EigenDA V1)")
-		}
-
-		data, err := m.eigenda.Get(ctx, versionedCert.SerializedCert)
-		if err == nil {
-			err = m.eigenda.Verify(ctx, versionedCert.SerializedCert, data)
-			if err != nil {
-				return nil, fmt.Errorf("verify EigenDA V1 cert: %w", err)
-			}
-			return data, nil
-		}
-
-		return nil, fmt.Errorf("get from EigenDA V1 backend: %w", err)
-
-	case certs.V1VersionByte, certs.V2VersionByte:
-		// The cert must be verified before attempting to get the data, since the GET logic
-		// assumes the cert is valid. Verify v2 doesn't require a payload
-		// because the payload is checked inside the Get function below.
-		err := m.eigendaV2.VerifyCert(ctx, versionedCert, opts.L1InclusionBlockNum)
-		if err != nil {
-			return nil, fmt.Errorf("verify EigenDACert: %w", err)
+func (m *EigenDAManager) backupToSecondary(ctx context.Context, commitment []byte, value []byte) {
+	if m.secondary.AsyncWriteEntry() { // publish put notification to secondary's subscription on PutNotify topic
+		m.log.Debug("Publishing data to async secondary stores", "commitment", commitment)
+		m.secondary.Topic() <- secondary.PutNotify{
+			Commitment: commitment,
+			Value:      value,
 		}
-
-		m.log.Debug("Reading blob from EigenDAV2 backend", "returnEncodedPayload", opts.ReturnEncodedPayload)
-		payloadOrEncodedPayload, err := m.eigendaV2.Get(ctx, versionedCert, opts.ReturnEncodedPayload)
+		// secondary is available only for synchronous writes
```

Comment:
I'm not understanding what this comment is trying to say. Why is it at the bottom of the `if` statement?

Reply:
wasn't my code just moved it around. But I think this is meant to describe the else branch. Moved it next to the else statement: [b233f37](https://github.com/Layr-Labs/eigenda/pull/1914/commits/b233f37c568a574c9cad8b7e44bba5562a538097)